File: Algorithms/Python3.x/1200-Minimum_Absolute_Difference.py
# Runtime: 468 ms, faster than 17.07% of Python3 online submissions for Minimum Absolute Difference.
# Memory Usage: 28 MB, less than 100.00% of Python3 online submissions for Minimum Absolute Difference.
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        result = []
        min_diff = float("inf")
        for i in range(1, len(arr)):
            if arr[i] - arr[i - 1] < min_diff:
                result = [[arr[i - 1], arr[i]]]
            elif arr[i] - arr[i - 1] == min_diff:
                result.append([arr[i - 1], arr[i]])
            min_diff = min(arr[i] - arr[i - 1], min_diff)
        return result

File: Algorithms/C++/1200-Minimum_Absolute_Difference.cpp
// Runtime: 208 ms, faster than 5.35% of C++ online submissions for Minimum Absolute Difference.
// Memory Usage: 26.3 MB, less than 100.00% of C++ online submissions for Minimum Absolute Difference.
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        int min_diff = INT_MAX;
        sort(arr.begin(), arr.end());
        vector<vector<int>> result;
        for (int i = 1; i < arr.size(); ++i) {
            vector<int> tmp = {arr[i - 1], arr[i]};
            int diff = tmp[1] - tmp[0];
            if (diff < min_diff) {
                result.clear();
                result.push_back(tmp);
            } else if (diff == min_diff) {
                result.push_back(tmp);
            }
            min_diff = min(min_diff, diff);
        }
        return result;
    }
};
